pkg/tools/sandbox/restartable_server.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sandbox
+
+import (
+	"context"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/networkservicemesh/sdk/pkg/tools/grpcutils"
+)
+
+type restartableServer struct {
+	ctx, currentCtx context.Context
+	cancelCurrent   context.CancelFunc
+
+	startFunction func(context.Context)
+}
+
+func newRestartableServer(
+	ctx context.Context,
+	t *testing.T,
+	serveURL *url.URL,
+	startFunction func(ctx context.Context),
+) *restartableServer {
+	r := &restartableServer{
+		ctx:           ctx,
+		cancelCurrent: func() {},
```

Comment:
Why is this empty? Is it means that we'll not able to restart a new server?

Reply:
No, it actually means that we are **able** to restart a new server, and it wouldn't make anything on server stop (because it is not yet running).